drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt

```diff
@@ -81,172 +91,185 @@ class PaymentMethodAdapter @JvmOverloads constructor(
+        private val binding: PaymentMethodsListItemBinding,
+        private val imageLoader: ImageLoader
+    ) : BaseViewHolder(binding) {
+        fun bind(model: GiftCardPaymentMethodModel) = with(binding) {
+            val context = binding.root.context
+            textViewText.text = context.getString(R.string.card_number_4digit, model.lastFour)
+            imageLoader.load(model.imageId, imageViewLogo)
+            if (model.transactionLimit == null || model.shopperLocale == null) {
+                textViewDetail.isVisible = false
+            } else {
+                val value = CurrencyUtils.formatAmount(
+                    model.transactionLimit,
+                    model.shopperLocale
+                )
+                textViewDetail.apply {
+                    isVisible = true
+                    text = context.getString(R.string.checkout_giftcard_max_transaction_limit, value)
+                }
+            }
+            if (model.amount == null || model.shopperLocale == null) {
+                textViewEndText.isVisible = false
+            } else {
+                val value = CurrencyUtils.formatAmount(
+                    model.amount,
+                    model.shopperLocale
+                )
+                textViewEndText.apply {
+                    isVisible = true
+                    text = context.getString(R.string.checkout_negative_amount, value)
+                }
+            }
+            itemView.setOnClickListener(null)
+        }
     }
 
-    private fun onHeaderActionClick(header: PaymentMethodHeader) {
-        onPaymentMethodSelectedCallback?.onHeaderActionSelected(header)
+    class HeaderVH(private val binding: PaymentMethodsListHeaderBinding) : BaseViewHolder(binding) {
+        fun bind(
+            model: PaymentMethodHeader,
+            onPaymentMethodSelectedCallback: OnPaymentMethodSelectedCallback?
+        ) = with(binding) {
+            paymentMethodHeaderTitle.setText(model.titleResId)
+            if (model.actionResId == null) {
+                paymentMethodHeaderAction.isVisible = false
+            } else {
+                paymentMethodHeaderAction.apply {
+                    isVisible = true
+                    setText(model.actionResId)
+                    setOnClickListener {
+                        onPaymentMethodSelectedCallback?.onHeaderActionSelected(model)
+                    }
+                }
+            }
+        }
     }
 
-    private fun getView(parent: ViewGroup, id: Int): View {
-        return LayoutInflater.from(parent.context).inflate(id, parent, false)
+    class NoteVH(private val binding: PaymentMethodsListNoteBinding) : BaseViewHolder(binding) {
+        fun bind(model: PaymentMethodNote) = with(binding) {
+            paymentMethodNote.text = model.note
+        }
     }
 
-    companion object {
-        private val TAG = LogUtil.getTag()
-    }
+    open class BaseViewHolder(binding: ViewBinding) : RecyclerView.ViewHolder(binding.root)
```

Comment:
This class is useless, might as well replace it with just `RecyclerView.ViewHolder`

Reply:
agree